components/gif_picker/components/Categories/index.tsx

```diff
@@ -2,23 +2,27 @@
 // See LICENSE.txt for license information.
 
 import React, {PureComponent} from 'react';
-import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
 
 import {requestCategoriesList, requestCategoriesListIfNeeded, saveSearchBarText, saveSearchScrollPosition, searchTextUpdate} from 'mattermost-redux/actions/gifs';
 
+import {GlobalState} from 'types/store';
+import {GfycatAPIItem, GfycatAPITag} from 'types/external/gfycat';
+
 import {trackEvent} from 'actions/telemetry_actions.jsx';
-import * as PostUtils from 'utils/post_utils';
+
+import {getImageSrc} from 'utils/post_utils';
 
 import InfiniteScroll from 'components/gif_picker/components/InfiniteScroll';
+import {appProps} from 'components/gif_picker/gif_picker';
 
 import './Categories.scss';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     return {
-        ...state.entities.gifs.categories,
-        ...state.entities.gifs.cache,
-        appProps: state.entities.gifs.app,
+        tagsList: state.entities.gifs.categories.tagsList,
```

Comment:
I had to explicitly mention key such as `tagsList` because TypeScript was not automatically inferring types from `state.entities.gifs.categories` and `state.entities.gifs.cache`. The reason for that type for `state.entities.gifs.categories` and `state.entities.gifs.cache` was typed as `any` in `GlobalState`.

Reply:
It might actually be good to add types for that state as well. I don't think it should've been left as `any` to begin with. I imagine it was probably skipped because the gif picker code mostly wasn't written by us

Reply:
The entire `state.entities.gifs` was typed as `any`. I added types for all under `packages/types/src/gifs.ts`. 

I also inserted external types in the same file. Because I had to use some of them, such as `GfycatAPIItem`. I couldn't import `GfycatAPIItem` when they are under `types/external`. Please let me know if it is OK?

Reply:
Yeah, that should be fine. I forgot the state for gifs were stored in `state.entities` and used in the types package, so that's probably why putting them in the web app's `types/external` wasn't allowed.

Thanks for defining all those. I didn't realize they had so many fields, but I guess that's what you get when using someone else's API

Reply:
> Yeah, that should be fine. I forgot the state for gifs were stored in `state.entities` and used in the types package, so that's probably why putting them in the web app's `types/external` wasn't allowed.

You are probably right. 

> Thanks for defining all those. I didn't realize they had so many fields, but I guess that's what you get when using someone else's API.

Exactly